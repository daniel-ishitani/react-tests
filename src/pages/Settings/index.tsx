import React from 'react';

import Card from '../../components/Card';
import Form from '../../components/Form';
import Footer from '../../components/Footer';
import ParticipantList from '../../components/ParticipantList';

const Settings = () => {
  return (
    <Card>
      <section>
        <h1>Let's start!</h1>
        <Form />
        <ParticipantList />
        <Footer />
      </section>
    </Card>
  )
};

export default Settings;
