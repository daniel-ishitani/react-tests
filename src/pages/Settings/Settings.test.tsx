import React from 'react';
import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import Settings from '.';

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

describe('Settings', () => {
  it('renders correctly', () => {
    const { container } = render(
      <RecoilRoot>
        <Settings />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
