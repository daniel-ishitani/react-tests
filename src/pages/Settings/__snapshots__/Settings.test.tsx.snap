// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h1>
        Let's start!
      </h1>
      <form>
        <div
          class="inputBtnGroup"
        >
          <input
            placeholder="Insert the participants names"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Add
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="footer"
      >
        <button
          class="button"
          disabled=""
        >
          Start
        </button>
        <img
          alt="Shopping bags"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
